src/hooks/useToggle.ts

```diff
@@ -0,0 +1,12 @@
+import { useCallback, useState } from 'react'
+
+export const useToggle = (initialState = false) => {
+  // Initialize the state
+  const [state, setState] = useState(initialState)
+
+  // Define and memorize toggler function in case we pass down the comopnent,
+  // This function change the boolean value to it's opposite value
+  const toggle = useCallback(() => setState((state) => !state), [])
+
+  return [state, toggle] as const
```

Comment:
```suggestion
  return [state, toggle]
```
I don't think we need this

Reply:
we do, for types

Reply:
Type inference works just fine here.  Are you sure? 
Without `as const`
![image](https://user-images.githubusercontent.com/51168865/134502538-f7c48ec0-faa7-449f-b16f-0b220f5bba2e.png)
With `as const`
![image](https://user-images.githubusercontent.com/51168865/134502723-0aadd234-1654-4d70-9c3d-eb2b436ce2c5.png)
Which is basically the same.

Anyway, why do we need this hook in the first place? It seems that it's not used at all.

Reply:
not the same with `as const` first item of the array is a boolean and second item is a fn always; without `as const` type is an array of any size can have items booleans or fns.

But yes this might get removed or not depending on another PR that builds on this one

Reply:
Just tested this. My mistake, you're right about `as const` 👍🏻

Reply:
basically without `as const` it's a regular array, and with `as const` it's a **tuple**(array with a fixed number of elements)

Reply:
Yep, that the first element is type boolean and the second of type fn which is important.

Another way to do it is to define the return type of the fn but `as const` it's just convenient